src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1721,11 +1721,13 @@ tx_gateway_frontend::do_end_txn(
         }
         bool is_status_ok = false;
         if (is_fetch_tx_supported()) {
-            is_status_ok = tx.status == tm_transaction::tx_status::ongoing
+            is_status_ok = tx.status == tm_transaction::tx_status::ready
```

Comment:
it's gonna be a cleaner fix if we replace `unknown_server_error` with `invalid_txn_state` in [this line](https://github.com/redpanda-data/redpanda/pull/7996/files#diff-4668c1f5a71baddf4be6a01da9ed359a1092d7dd8b83c2cb655175d7eab71b78R1761)

Reply:
hyperlink doesn't seem valid.

Reply:
It points to this piece at lines 1758-1759.

```
outcome->set_value(tx_errc::unknown_server_error);
co_return tx_errc::unknown_server_error;
```

Have you changed `outcome->set_value(...)` when you tried it first?

Reply:
> It points to this piece at lines 1758-1759.

Oh, looks like the hyperlink only works if that line is expanded/visible in the github UI. weird.

> when you tried it first?

you mean when I wrote a quick fix to test the change? Then I added a separate if condition for `ready` state and set the outcome/return.